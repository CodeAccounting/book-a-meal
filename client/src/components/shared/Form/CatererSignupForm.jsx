import React, { Fragment } from 'react';
import { RenderInput } from '../../shared/FormComponents';
import { formPropTypes } from '../../../helpers/proptypes';

/**
 * @exports
 * @function CatererSignupForm
 * @param {object} props
 * @returns {JSX} CatererSignupForm
 */
const CatererSignupForm = ({ state, handlers }) => (
  <Fragment>
    <RenderInput
      type="text"
      name="businessName"
      id="businessName"
      label="Business Name"
      required
      value={state.values.businessName}
      placeholder=""
      handleChange={handlers.handleChange}
      handleBlur={handlers.handleBlur}
      handleFocus={handlers.handleFocus}
      meta={{
        touched: state.touched.businessName,
        error: state.error.businessName,
        asyncValidating: state.asyncValidating
      }}
    />
    <RenderInput
      type="email"
      name="email"
      label="Email Address"
      id="email"
      required
      value={state.values.email}
      placeholder=""
      handleChange={handlers.handleChange}
      handleBlur={handlers.handleBlur}
      handleFocus={handlers.handleFocus}
      meta={{
          touched: state.touched.email,
          error: state.error.email,
          asyncValidating: state.asyncValidating
        }}
    />
    <RenderInput
      type="password"
      name="password"
      label="Password"
      id="password"
      required
      value={state.values.password}
      placeholder=""
      handleChange={handlers.handleChange}
      handleBlur={handlers.handleBlur}
      handleFocus={handlers.handleFocus}
      meta={{
          touched: state.touched.password,
          error: state.error.password
        }}
    />
    <RenderInput
      type="password"
      name="passwordConfirm"
      label="Password Confirm"
      id="passwordConfirm"
      required
      value={state.values.passwordConfirm}
      placeholder=""
      handleChange={handlers.handleChange}
      handleBlur={handlers.handleBlur}
      handleFocus={handlers.handleFocus}
      meta={{
          touched: state.touched.passwordConfirm,
          error: state.error.passwordConfirm
        }}
    />
    <RenderInput
      type="text"
      name="businessPhoneNo"
      id="businessPhoneNo"
      label="Business Phone Number (e.g. 08123456789)"
      required
      value={state.values.businessPhoneNo}
      placeholder=""
      handleChange={handlers.handleChange}
      handleBlur={handlers.handleBlur}
      handleFocus={handlers.handleFocus}
      meta={{
        touched: state.touched.businessPhoneNo,
        error: state.error.businessPhoneNo,
        asyncValidating: state.asyncValidating
      }}
    />
    <RenderInput
      type="text-area"
      row={2}
      name="businessAddress"
      id="businessAddress"
      label="Business Address"
      required
      value={state.values.businessAddress}
      placeholder=""
      handleChange={handlers.handleChange}
      handleBlur={handlers.handleBlur}
      handleFocus={handlers.handleFocus}
      meta={{
        touched: state.touched.businessAddress,
        error: state.error.businessAddress,
        asyncValidating: state.asyncValidating
      }}
    />
  </Fragment>
);

CatererSignupForm.propTypes = {
  ...formPropTypes('catererSignup')
};

export default CatererSignupForm;
