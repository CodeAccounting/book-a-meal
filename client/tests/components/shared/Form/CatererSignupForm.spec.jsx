import React from 'react';
import CatererSignupForm from '../../../../src/app/shared/Form/CatererSignupForm';
import { formComponentSetup } from '../../../../tests/setup/formSetup';

describe('CatererSignupForm', () => {
  afterAll(() => {
    jest.clearAllMocks();
  });

  it('renders correctly', () => {
    const { handlers, state } = formComponentSetup('catererSignup');
    const wrapper = shallow(<CatererSignupForm type="catererSignup" state={state} handlers={handlers} />);

    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
