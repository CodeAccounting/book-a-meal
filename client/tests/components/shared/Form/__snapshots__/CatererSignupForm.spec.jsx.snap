// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CatererSignupForm renders correctly 1`] = `
<React.Fragment>
  <RenderInput
    handleBlur={[MockFunction]}
    handleChange={[MockFunction]}
    handleFocus={[MockFunction]}
    id="businessName"
    label="Business Name"
    meta={
      Object {
        "asyncValidating": false,
        "error": null,
        "touched": false,
      }
    }
    name="businessName"
    placeholder=""
    required={true}
    type="text"
    value=""
  />
  <RenderInput
    handleBlur={[MockFunction]}
    handleChange={[MockFunction]}
    handleFocus={[MockFunction]}
    id="email"
    label="Email Address"
    meta={
      Object {
        "asyncValidating": false,
        "error": null,
        "touched": false,
      }
    }
    name="email"
    placeholder=""
    required={true}
    type="email"
    value=""
  />
  <RenderInput
    handleBlur={[MockFunction]}
    handleChange={[MockFunction]}
    handleFocus={[MockFunction]}
    id="password"
    label="Password"
    meta={
      Object {
        "error": null,
        "touched": false,
      }
    }
    name="password"
    placeholder=""
    required={true}
    type="password"
    value=""
  />
  <RenderInput
    handleBlur={[MockFunction]}
    handleChange={[MockFunction]}
    handleFocus={[MockFunction]}
    id="passwordConfirm"
    label="Password Confirm"
    meta={
      Object {
        "error": null,
        "touched": false,
      }
    }
    name="passwordConfirm"
    placeholder=""
    required={true}
    type="password"
    value=""
  />
  <RenderInput
    handleBlur={[MockFunction]}
    handleChange={[MockFunction]}
    handleFocus={[MockFunction]}
    id="businessPhoneNo"
    label="Business Phone Number (e.g. 08123456789)"
    meta={
      Object {
        "asyncValidating": false,
        "error": null,
        "touched": false,
      }
    }
    name="businessPhoneNo"
    placeholder=""
    required={true}
    type="text"
    value=""
  />
  <RenderInput
    handleBlur={[MockFunction]}
    handleChange={[MockFunction]}
    handleFocus={[MockFunction]}
    id="businessAddress"
    label="Business Address"
    meta={
      Object {
        "asyncValidating": false,
        "error": null,
        "touched": false,
      }
    }
    name="businessAddress"
    placeholder=""
    required={true}
    row={2}
    type="text-area"
    value=""
  />
</React.Fragment>
`;
